src/server/rpc/procedures/ConnectN/ConnectN.js

```diff
@@ -0,0 +1,365 @@
+    debug = require('debug'),
+    log = debug('NetsBlox:RPCManager:ConnectN:log'),
+    trace = debug('NetsBlox:RPCManager:ConnectN:trace'),
+    Constants = require('../../../../common/Constants'),
+    info = debug('NetsBlox:RPCManager:ConnectN:info');
+
+/**
+ * ConnectNRPC - This constructor is called on the first request to an RPC
+ * from a given room
+ *
+ * @constructor
+ * @return {undefined}
+ */
+var ConnectNRPC = function() {
+    this.board = ConnectNRPC.getNewBoard();
+    this._winner = null;
+    this.lastMove = null;
+};
+
+/**
+ * Return the path to the given RPC
+ *
+ * @return {String}
+ */
+ConnectNRPC.getPath = function() {
+    return '/connectn';
+};
+
+/**
+ * This function is used to expose the public API for RPC calls
+ *
+ * @return {Array<String>}
+ */
+ConnectNRPC.getActions = function() {
+    return [
+            'newGame',  // Clear the board
+            'play',  // Play a tile at the given location
+            'isOpen',  // for testing
+            ];
+};
+
+// Actions
+ConnectNRPC.prototype.newGame = function(req, res) {
+    var roleId = req.netsbloxSocket.roleId;
+    this.numRow = req.query.row;
+    this.numCol = req.query.column;
+    this.numDotsToConnect = req.query.numDotsToConnect;
+    this._winner = null;
+    this.lastMove = null;
+    this.board = ConnectNRPC.getNewBoard(this.numRow, this.numCol);
+    info(req.query.roleId+' is clearing board and creating a new one with size: ', this.numRow, ", ", this.numCol);
+    //console.log("ConnectN inside Clear, print of req.query: ", req.query, " numDotsToConnect", this.numDotsToConnect);
+    //console.log("new board print out: ",  this.board);
+
+
+      req.netsbloxSocket._room.sockets()
+          .filter(s => s !== req.netsbloxSocket)
+          .forEach(socket => socket.send({
+                  type: 'message',
+                  msgType: 'start',
+                  dstId: Constants.EVERYONE,
+                  content: {
+                      row: this.numRow,
+                      column: this.numCol,
+                      numDotsToConnect: this.numDotsToConnect
+                  }
+              }));
+
+
+    res.status(200).send(true);
+};
+
+
+ConnectNRPC.prototype.isOpen = function(req, res) {
+    var row = req.query.row-1,
+        column = req.query.column-1,
+        isValid = [row, column].map(n => n > -1 && n < 3),
+        open;
+
+    isValid = [row, column].map(n => n > -1 && n < 3);
+    if (isValid[0] && isValid[1]) {
+        open = this.board[row][column] === null;
+        return res.send(open);
+    }
+
+    return res.status(400).send('bad position');
+};
+
+ConnectNRPC.prototype.play = function(req, res) {
+    //console.log("ConnectN inside play, req.query: ", req.query)
+    var row = req.query.row,
+        column = req.query.column,
+        roleId = req.netsbloxSocket.roleId,
+        open = this.board[row][column] === null,
+        isOnBoard = [row, column].every(this.isValidPosition.bind(this));
```

Comment:
This will need to be updated; it looks like `isValidPosition` assumes that it is a square board (which it may not be)